Fixes #348 resizes private volume sizes of sd-app sd-log

kushaldas opened the pull request:
sd-app:private is now 10GB
sd-log:private is now 5GB

## How to test?

- [ ] Clone the repo to `dom0` and get the config.json and sd-journalist.sec file for your installation
- [ ] Update config.json to have the vmsizes, 

```
"vmsizes": {
    "sd_app": 10,
    "sd_log": 5
  },
```
- [ ] run `make sd-app`
- [ ] run `make sd-log`
- [ ] run `make test`
- [ ] Now, manually change the size for `sd-app` to 11GB `qvm-volume resize sd-app:private 11GiB
- [ ] Run `make sd-app` again, this should fail with a proper failure message.

 All of these commands should execute properly.

If you can check the private volume size of the `sd-app` and `sd-log` vm in the GUI too.

emkll commented on tests/test_vms_exist.py:
```diff
@@ -176,6 +181,11 @@ def sd_log_template(self):
+        # Should be 5GB
+        # >>> 1024 * 1024 * 5 * 1024
+        vol = vm.volumes["private"]
+        self.assertEqual(vol.size, 5368709120)
```
Here, you are testing the template vm, not the AppVM (sd-log). Strangely though, this test is not failing. Perhaps it is not running?

emkll commented on dom0/sd-svs.sls:
```diff
@@ -45,6 +45,14 @@ sd-svs:
+sd-svs-private-volume-size:
+  cmd.run:
+    - name: >
+        qvm-volume resize sd-svs:private 10GiB
```
:heavy_check_mark: It seems like qvm.volume is not implemented in salt, so we must use cmd.run here

emkll commented:
Due to changes introduced in #407, this will also need a rebase on latest master: sd-svs -> sd-app

emkll commented on scripts/validate-config:
```diff
@@ -88,6 +90,28 @@ class SDWConfigValidator(object):
+
+        app = Qubes()
+        if "sd-svs" in app.domains:
+            vm = app.domains["sd-svs"]
```
same as comment from L:101

kushaldas replied:
ah, I will have to update this.

emkll commented on scripts/validate-config:
```diff
@@ -88,6 +90,28 @@ class SDWConfigValidator(object):
+        """This method checks for existing private volume size and new
+        values in the config.json"""
+        assert "vmsizes" in self.config
+        assert "sd_svs" in self.config["vmsizes"]
```
same as comment from L:101

emkll commented on scripts/validate-config:
```diff
@@ -88,6 +90,28 @@ class SDWConfigValidator(object):
+            vm = app.domains["sd-svs"]
+            vol = vm.volumes["private"]
+            assert (
+                vol.size <= self.config["vmsizes"]["sd_svs"] * 1024 * 1024 * 1024
```
same as comment from L:101

kushaldas commented on config.json.example:
```diff
@@ -3,5 +3,9 @@
-  }
+  },
+  "vmsizes": {
+        "sd_app": 10,
```
I will update this.

kushaldas replied:
Be

> nit: is there a reason the underscore here is required? Intuitively it sounds like having the actual vm name would be less-error prone here (`sd-app`, `sd-log`)

Because Python variables can not have `-` in the name. This is used in the Jinja template to access the config.json values.

Errors which we will see otherwise:

```
    Set up dom0 config files, including RPC policies, and create VMs

    [ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'sd_app'

    [CRITICAL] Rendering SLS 'base:sd-app' failed: Jinja variable 'dict object' has no attribute 'sd_app'

    [ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'sd_log'

    [CRITICAL] Rendering SLS 'base:sd-log' failed: Jinja variable 'dict object' has no attribute 'sd_log'

    local:

        Data failed to compile:

    ----------

        Rendering SLS 'base:sd-app' failed: Jinja variable 'dict object' has no attribute 'sd_app'

    ----------

        Rendering SLS 'base:sd-log' failed: Jinja variable 'dict object' has no attribute 'sd_log'
```

conorsch commented on config.json.example:
```diff
@@ -3,5 +3,9 @@
     "key": "Il8Xas7uf6rjtc0LxYwhrx"
-  }
+  },
+  "vmsizes": {
```
No, because it's JSON. =/ If/when this file becomes YAML, then we can add comments. The format question is best addressed in the context of #51, so recommend tolerating the lack of comments for now.

emkll commented:
rebased on latest `master`

kushaldas commented:
@emkll thank you for the comment, i will update it tomorrow in my time.

eloquence commented on scripts/validate-config:
```diff
@@ -88,6 +90,28 @@ class SDWConfigValidator(object):
+            vol = vm.volumes["private"]
+            assert (
+                vol.size <= self.config["vmsizes"]["sd_app"] * 1024 * 1024 * 1024
+            ), "sd-svs private volume is already bigger than configuration."
```
sd-app

kushaldas commented:
>[1]. We should either validate (or round) the value in the config, or document this constraint. Since we cannot add comments in the json file, so it may be best if we addressed this in the validator. What do you think?

I have added a validation check for integer values of the private volume sizes.
